File: vendas-services/src/main/java/br/com/vendas/services/order/pricetable/PriceTableService.java
package br.com.vendas.services.order.pricetable;

import java.util.List;

import br.com.vendas.domain.customer.Customer;
import br.com.vendas.domain.order.PriceTable;
import br.com.vendas.exception.RegistrationException;
import br.com.vendas.services.support.ServiceResponse;

public interface PriceTableService {
	
	/**
	 * Retorna todas as tabelas de preço de determinada filial
	 * @param empresaID
	 * @param filialID
	 * @return
	 */
	ServiceResponse<List<PriceTable>> findAllByBranche(Integer organizationID, Integer branchID);
	
	/**
	 * Salva a tabela de preço
	 * @param priceTable
	 * @return
	 */
	ServiceResponse<PriceTable> save( Integer userID, PriceTable priceTable );
	
	/**
	 * Atualiza a tabela de preço
	 * @param priceTable
	 * @return
	 * @throws RegistrationException
	 */
	ServiceResponse<PriceTable> saveOrUpdate( Integer userID, PriceTable priceTable ) ;
	
	
	void save(List<PriceTable> priceTables);
	

	/**
	 * Retorna todos os registros com data de alteração maior que a data passada por parametro.
	 * @param date
	 * @return
	 */
	ServiceResponse<List<PriceTable>> findAllByChangeGreaterThan( Long date, Integer organizationID, Integer offset );

}

File: vendas-parent/vendas-web/src/main/webapp/app/js/ng/services/PriceTableService.js
'use strict';

vendasApp.factory('PriceTableService',function(Restangular){
	
	var priceTableEdit; 
	return {
		
		save : function(priceTable, userID){
			var priceTable_ = Restangular.all("private").all("v1").all('priceTable').all('save').post(priceTable ,{},{'userID' : userID});
			return priceTable_;
        },
		
		
		  /**
         * Retorna todas as tabelas de preço de determinada filial.
         * @returns
         */
        getAllByBranch: function(organizationID,branchID){
        	
        	var parameters = {
					'organizationID' : organizationID, 
					'branchID' : branchID					
					};
        	
        	
        	return Restangular.all("private").all("v1").all("priceTable").all("findAllByBranche").getList(parameters).then(function(result){
				var p = {};
				p.value = result.value;
				p.rowCount = result.rowCount;			
				return p;
			});
        }, 
        
        
        setPriceTableEdition : function(priceTable){
        	priceTableEdit = priceTable;
        },
        
        getPriceTableEdition : function(){
        	return priceTableEdit;
        } 
		
	};
});